### PR TITLE
Use XCCDF 1.2 to generate STIG HTML tables

### PR DESCRIPTION
**Pull request by jan-cerny**
#### Description:
We will use Jinja 2 instead of XSLT transformations to generate STIG HTML tables.

We will use ssg-${PRODUCT}-xccdf-1.2.xml to generate table-${PRODUCT}-stig.html. Note that the stig_overlay.xml file remains in the XCCDF 1.1 namespace, but the temporary internal file unlinked-stig-xccdf.xml will now be in XCCDF 1.2 namespace.

This PR supersedes PR https://github.com/ComplianceAsCode/content/pull/9322. The reason for replacing the PR by new PR is that the previous PR tried to modify the XSLT template, but it would break the manual STIG table and it wasn't easy to adapt the template to work with XML files which will now be each in a different namespace.

For more details, please read commit messages of each commit.

#### Rationale:
This reduces our dependency on XCCDF 1.1 so it will help us to remove the XCCDF 1.1 format in future.

### CODE REVIEW

File: cmake/SSGCommon.cmake [hunk 1]
```diff
@@ -1152,19 +1151,18 @@ macro(ssg_build_html_stig_tables PRODUCT)
     )
     add_custom_command(
         OUTPUT "${CMAKE_CURRENT_BINARY_DIR}/unlinked-stig-xccdf.xml"
-        COMMAND "${XSLTPROC_EXECUTABLE}" -stringparam overlay "${CMAKE_BINARY_DIR}/${PRODUCT}/overlays/stig_overlay.xml" --stringparam ocil-document "${CMAKE_BINARY_DIR}/${PRODUCT}/ocil-linked.xml" --output "${CMAKE_CURRENT_BINARY_DIR}/unlinked-stig-xccdf.xml" "${CMAKE_CURRENT_SOURCE_DIR}/transforms/xccdf-apply-overlay-stig.xslt" "${CMAKE_BINARY_DIR}/ssg-${PRODUCT}-xccdf.xml"
+        COMMAND "${XSLTPROC_EXECUTABLE}" -stringparam overlay "${CMAKE_BINARY_DIR}/${PRODUCT}/overlays/stig_overlay.xml" --stringparam ocil-document "${CMAKE_CURRENT_BINARY_DIR}/ocil-linked.xml" --output "${CMAKE_CURRENT_BINARY_DIR}/unlinked-stig-xccdf.xml" "${CMAKE_CURRENT_SOURCE_DIR}/transforms/xccdf-apply-overlay-stig.xslt" "${CMAKE_BINARY_DIR}/ssg-${PRODUCT}-xccdf-1.2.xml"
```
Comment (yuumasato) [inline cmake/SSGCommon.cmake:1154]: Maybe you should add a `DEPENDS "${CMAKE_CURRENT_BINARY_DIR}/ocil-linked.xml"`?

File: utils/tables/stig_template.html [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+	<td>{{{ rule["SRG"] }}}</td>
+	<td><pre>{{{ rule["description"] }}}</pre></td>
+	<td><pre>{{{ rule["check"] }}}</pre></td>
+	<td><pre>{{{ rule["fixtext"] }}}</pre></td>
+	<td>{{{ rule["version"] }}}</td>
+	<td>{{{ rule["mapped_rule"] }}}</td>
+</tr>
+{{% endfor %}}
+</table>
+{{% endblock %}}
```
Comment (yuumasato) [inline utils/tables/stig_template.html:58]: `newline` missing at the end

**Review by yuumasato — CHANGES_REQUESTED** [review]

File: utils/tables/stig_template.html [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+	</thead>
+{{% for rule in rules %}}
+<tr>
+	<td>{{{ rule["V-ID"] }}}</td>
+	<td>{{{ rule["CCI"] }}}</td>
+	<td>{{{ rule["CAT"] }}}</td>
+	<td>{{{ rule["title"] }}}</td>
+	<td>{{{ rule["SRG"] }}}</td>
+	<td><pre>{{{ rule["description"] }}}</pre></td>
+	<td><pre>{{{ rule["check"] }}}</pre></td>
```
[the remark this answers is not in the document]
Comment (jan-cerny) [reply utils/tables/stig_template.html:51]: This is something from the category "works fine on my machine" :smile: :smile: , I hope that adding the dependency to CMake will help.

[the remark this answers is not in the document]
Comment (yuumasato) [reply utils/tables/stig_template.html:51]: Works on my machine now, :smile:

**Review by yuumasato — APPROVED** [review]

**Pull request merged**